### PR TITLE
285 use tensorflow23

### LINKED ISSUE #285: Use tensorflow 2.3
# Issue description

The latest stable TensorFlow is 2.3 now.

We should update to this version and check if everything still works.

## Type of Issue

- [ ] New feature request

### PR DESCRIPTION
**Pull request by mathpluscode**
# Description

- Update tensorflow requirement to 2.3
- batch size is the batch size per device now
- profiler tested locally without GPU

Fixes #285

## Type of change

- [ ] New feature

# How Has This Been Tested?

- Tested locally without CPU, train/predict works, tests passed
- Tested on PT machine with one GPU, train/predict works
- Tested on PT machine with two GPUs, train works

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by mathpluscode** [comment]
Wait a sec, https://anaconda.org/anaconda/tensorflow-gpu/labels conda is not supporting tensorflow 2.3 yet :(
